test/suites/tuned/microshift-tuned.robot

```diff
@@ -0,0 +1,97 @@
+
+    ${old_hash}=    Command Should Work    cat /var/lib/microshift-tuned.yaml
+    Command Should Work
+    ...    sed -i 's/^offline_cpu_set=.*$/offline_cpu_set=/' /etc/tuned/microshift-baseline-variables.conf
+    Restart MicroShift-Tuned Expecting Reboot
+    ${new_hash}=    Command Should Work    cat /var/lib/microshift-tuned.yaml
+    Should Not Be Equal    ${old_hash}    ${new_hash}
+
+No Reboot If Not Allowed
+    [Documentation]    If reboot_after_apply is False, then microshift-tuned should not reboot the host even if the profile changed.
+
+    Command Should Work
+    ...    sed -i 's/^offline_cpu_set=.*$/offline_cpu_set=3/' /etc/tuned/microshift-baseline-variables.conf
+    Command Should Work    sed -i 's/reboot_after_apply:.*/reboot_after_apply: False/' /etc/microshift/tuned.yaml
+    Restart MicroShift-Tuned Not Expecting Reboot
+
+Can Activate Any Tuned Profile
+    [Documentation]    Verify that microshift-tuned will activate any given tuned profile.
+
+    Command Should Work    sed -i 's/profile:.*/profile: virtual-guest/' /etc/microshift/tuned.yaml
+    Restart MicroShift-Tuned Not Expecting Reboot
+    ${active}=    Command Should Work    tuned-adm active
+    Should Contain    ${active}    Current active profile: virtual-guest
+
+MicroShift-Tuned Requires Config To Function
+    [Documentation]    Verify that missing configuration will be fatal.
+    Command Should Work    rm /etc/microshift/tuned.yaml
+    Command Should Work    systemctl restart microshift-tuned.service
+    Sleep    5s
+    Systemctl Check Service SubState    microshift-tuned.service    failed
+
+
+*** Keywords ***
+Setup
+    [Documentation]    Setup test for the test suite
+    Login MicroShift Host
+    # We don't care for MicroShift when testing microshift-tuned.service
+    Stop MicroShift
+    Disable MicroShift
+    Command Should Work
+    ...    cp /etc/tuned/microshift-baseline-variables.conf /etc/tuned/microshift-baseline-variables.conf.bak
+    Command Should Work    cp /etc/microshift/tuned.yaml /etc/microshift/tuned.yaml.bak
+
+Teardown
+    [Documentation]    Teardown test after the test suite
+    Command Should Work
+    ...    mv /etc/tuned/microshift-baseline-variables.conf.bak /etc/tuned/microshift-baseline-variables.conf
+    Command Should Work    mv /etc/microshift/tuned.yaml.bak /etc/microshift/tuned.yaml
+    Enable MicroShift
+    Restart MicroShift-Tuned Expecting Reboot
+    Logout MicroShift Host
+
+Restart MicroShift-Tuned Expecting Reboot
+    [Documentation]    TODO
+    ${bootid}=    Get Current Boot Id
+    Command Should Work    systemctl restart microshift-tuned.service
+    Sleep    30s
+
+    Wait Until Keyword Succeeds    5m    15s
+    ...    Is System Rebooted    ${bootid}
+
+Restart MicroShift-Tuned Not Expecting Reboot
+    [Documentation]    TODO
+    ${bootid}=    Get Current Boot Id
+    Command Should Work    systemctl restart microshift-tuned.service
+    Sleep    60s
```

Comment:
Could we think of making this check a bit faster? Is there any other way we can get an indication that the microshift-tuned script "decided" not to reboot?

Reply:
I changed it but I'm not sure about robustness, we'll see